logbook-servlet/src/main/java/org/zalando/logbook/servlet/TeeResponse.java

```diff
@@ -71,14 +70,20 @@ public Origin getOrigin() {
     }
 
     @Override
-    public Multimap<String, String> getHeaders() {
-        final Multimap<String, String> headers = ArrayListMultimap.create();
+    public ListMultimap<String, String> getHeaders() {
+        this.headers = Optional.ofNullable(headers)
```

Comment:
> That would also tell you if the case is actually covered by tests :)

I re-checked and caching of headers is not required for any implementation of Raw/HttpRequest/Response, since the default implementation of Logbook and obfuscation can do that once and for all.